corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -149,7 +176,7 @@ hqDefine('data_dictionary/js/data_dictionary.js', function () {
         this.toggleGroup = function (group) {
             group.toggle();
             var groupIndex = _.findIndex(self.casePropertyList(), function (element) {
-                return element.name() === group.name();
+                return element.name === group.name;
```

Comment:
In the same case types, or for different case types? If the case type is different that is totally fine, if you mean two groups with the same name for the same case type, going forward, they would be treated as the same group. I am not sure what would happen at the point of creation, but for all future page loads, those groups would be merged into one. Given that name is the only thing that determines a group, it doesn't make sense to have two with the same name.